exclause/with.go

```diff
@@ -20,8 +20,9 @@ import (
 //	// WITH RECURSIVE `cte` AS (SELECT * FROM `users`) SELECT * FROM `cte`
 //	db.Clauses(exclause.With{Recursive: true, CTEs: []exclause.CTE{{Name: "cte", Subquery: exclause.Subquery{DB: db.Table("users")}}}}).Table("cte").Scan(&users)
 type With struct {
-	Recursive bool
-	CTEs      []CTE
+	Recursive    bool
+	Materialized bool
```

Comment:
Could you move the Materialized field to the CTE struct? It applies to individual CTEs, not the
entire WITH clause.